gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -140,23 +141,41 @@ private void writePolicyMetadata(final Map<String, PolicyMetadata> policyMetadat
     }
 
     /**
-     * This method finds policy dependencies from the bundle dependency graphf for a given policy id
+     * This method finds the policy dependencies from the bundle dependency graph for a given policy id
      * @param id String
      * @param rawBundle Bundle
      * @return Set
      */
-    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle) {
+    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle){
+        final Set<Dependency> dependencyList = new HashSet<>();
         Map<Dependency, List<Dependency>> dependencyListMap = rawBundle.getDependencyMap();
         if (dependencyListMap != null) {
-            Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
-            for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
-                Dependency parent = entry.getKey();
-                if (parent.getId().equals(id)) { // Search for "id" to get its dependencies
-                    return new HashSet<>(entry.getValue());
+            populateDependencies(dependencyListMap, id, dependencyList);
+        }
+        return dependencyList;
+    }
+
+    /**
+     * This method recursively loads the entity dependencies from the given entity id
```

Comment:
This needs modification if you are going to use the same method recursively only for Env entities.